reactive-streams-tck-tests/src/test/java/io/smallrye/mutiny/tcktests/MultiConcatMapNoPrefetchCompletionStageTckTest.java

```diff
@@ -0,0 +1,153 @@
+        CompletableFuture<Integer> three = new CompletableFuture<>();
+        CompletionStage<List<Integer>> result = Multi.createFrom().<CompletionStage<Integer>> items(one, two, three)
+                .onItem().transformToUni(i -> Uni.createFrom().completionStage(i))
+                .concatenate(false)
+                .collect().asList()
+                .subscribeAsCompletionStage();
+        one.complete(1);
+        two.complete(2);
+        three.complete(3);
+        Assert.assertEquals(Await.await(result), Arrays.asList(1, 2, 3));
+    }
+
+    @Override
+    public Flow.Publisher<Long> createFlowPublisher(long elements) {
+        return upstream(elements)
+                // Use supplyAsync on purpose to check the concurrency.
+                .onItem().transformToUni(i -> Uni.createFrom().completionStage(CompletableFuture.supplyAsync(() -> i)))
+                .concatenate(false);
+    }
+
+    @Override
+    public Flow.Publisher<Long> createFailedFlowPublisher() {
+        return failedUpstream()
+                // Use supplyAsync on purpose to check the concurrency.
+                .onItem().transformToUni(i -> Uni.createFrom().completionStage(CompletableFuture.supplyAsync(() -> i)))
+                .concatenate(false);
+    }
+
+    @Test
+    public void flatMapCsStageShouldMaintainOrderOfFuturesIfConcurrencyIs1() throws Exception {
+        CompletableFuture<Integer> one = new CompletableFuture<>();
+        CompletableFuture<Integer> two = new CompletableFuture<>();
+        CompletableFuture<Integer> three = new CompletableFuture<>();
+        CompletionStage<List<Integer>> result = Multi.createFrom().<CompletionStage<Integer>> items(one, two, three)
+                .onItem().transformToUni(i -> Uni.createFrom().completionStage(i)).concatenate(false)
+                .collect().asList()
+                .subscribeAsCompletionStage();
+        three.complete(3);
+        Thread.sleep(100L); //NOSONAR
+        two.complete(2);
+        Thread.sleep(100L); //NOSONAR
+        one.complete(1);
+        Assert.assertEquals(Await.await(result), Arrays.asList(1, 2, 3));
+    }
+
+    @Test
+    public void flatMapCsStageShouldOnlyMapOneElementAtATimeWithConcurrencyIs1() throws Exception {
+        CompletableFuture<Integer> one = new CompletableFuture<>();
+        CompletableFuture<Integer> two = new CompletableFuture<>();
+        CompletableFuture<Integer> three = new CompletableFuture<>();
+        AtomicInteger concurrentMaps = new AtomicInteger(0);
+        CompletionStage<List<Integer>> result = Multi.createFrom().<CompletionStage<Integer>> items(one, two, three)
+                .onItem().transformToUni(cs -> {
+                    Assert.assertEquals(1, concurrentMaps.incrementAndGet());
+                    return Uni.createFrom().completionStage(cs);
+                }).concatenate(false)
+                .collect().asList()
+                .subscribeAsCompletionStage();
+        Thread.sleep(100L); // NOSONAR
```

Comment:
We don't use Sonar anyway 😉

Reply:
Also would Awaitility be a better option that thread sleeps?

Reply:
Actually, it was a shameless copy from MultiFlatMapCompletionStageTckTest. I removed all thread sleeps.